app/src/main/java/org/standardnotes/notes/NoteActivity.kt

```diff
@@ -1,24 +1,98 @@
 package org.standardnotes.notes
 
+import android.animation.Animator
 import android.content.Intent
 import android.os.Bundle
 import android.support.v4.app.NavUtils
 import android.view.MenuItem
+import android.view.View
+import android.view.ViewAnimationUtils
+import android.view.ViewTreeObserver
 import org.standardnotes.notes.frag.NoteFragment
 
+
 class NoteActivity : BaseActivity() {
 
+    val REVEAL_ANIM_DURATION = 200L
+    var backPressed = false
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        overridePendingTransition(0, 0)
 
-        //title = note?.title ?: "New note"
         if (savedInstanceState == null) {
             val frag: NoteFragment = NoteFragment()
             frag.arguments = intent.extras
+
             supportFragmentManager.beginTransaction().replace(android.R.id.content, frag).commit()
+
+            if (frag.arguments == null) {
+                val rootView = findViewById(android.R.id.content)
+                val viewTreeObserver = rootView.viewTreeObserver
+                if (viewTreeObserver.isAlive) {
+                    rootView.visibility = View.INVISIBLE
+                    viewTreeObserver.addOnGlobalLayoutListener(object : ViewTreeObserver.OnGlobalLayoutListener {
+                        override fun onGlobalLayout() {
+                            circularReveal()
+                            rootView.viewTreeObserver.removeOnGlobalLayoutListener(this)
+                        }
+                    })
+                }
+            }
         }
-        supportActionBar?.setDisplayShowHomeEnabled(true)
-        supportActionBar?.setDisplayHomeAsUpEnabled(true)
+    }
+
+    override fun onBackPressed() {
+        if (!backPressed) {
+            overridePendingTransition(0, 0)
+            if (intent.extras == null) {
+                circularHide()
+                findViewById(android.R.id.content).postDelayed({
+                    super.onBackPressed()
+                }, REVEAL_ANIM_DURATION)
+            } else {
+                super.onBackPressed()
+            }
+        }
+        backPressed = true
+    }
+
+    private fun circularReveal() {
+        val rootView = findViewById(android.R.id.content)
+        val halfFab = 56.dpToPixels() / 2
+        val cx = rootView.width - resources.getDimension(R.dimen.activity_horizontal_margin).toInt() - halfFab
```

Comment:
I think we should be passing these coordinates through rather than calculating it here. This code is basically duplicating the layout information of a different activity, so if that changes then there is no link to update this.